lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) -> time depends on how long the string is.
+              #string of 3 chars will take less time than one of 100000 chars
+# Space Complexity: O(1) -> nothing new being created
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  array = string.upcase.split('')
+
+  punctuation = %w[. , / \ " ; : ! & ^ # $ @ % ( ) \ ]
+
+  punctuation.each do |i|
+    if array.include?(i)
+      array.delete(i)
+    end
+  end
+
+  while i <= array.length
+    return false if array[i] != array[-(i+1)]
+    i += 1
+  end 
+
+  return true
+  
 end
 
 # Time Complexity: ?
 # Space Complexity: ?
 def longest_prefix(strings)
```

Comment:
Your solution is actually O(n * m^2) of which n is the number of strings and m is the length of the shortest string.